waiter/src/waiter/scheduler.clj

```diff
@@ -730,13 +735,25 @@
 
 (defn add-instance-to-buffered-collection!
   "Helper function to add/remove entries into the transient store"
-  [transient-store max-instances-to-keep service-id instance-entry initial-value-fn remove-fn]
-  (swap! transient-store
-         (fn [service-id->failed-instances]
-           (update-in service-id->failed-instances [service-id]
-                      #(cond-> (or % (initial-value-fn))
-                         (= max-instances-to-keep (count %)) (remove-fn)
-                         true (conj instance-entry))))))
+  ([transient-store service-id instance-entry max-instances-to-keep initial-value-fn remove-fn]
+   (swap! transient-store
+          (fn [service-id->failed-instances]
+            (update-in service-id->failed-instances [service-id]
+                       #(cond-> (or % (initial-value-fn))
+                                (= max-instances-to-keep (count %)) (remove-fn)
+                                true (conj instance-entry))))))
+  ;; Since we have a map (failure-id to failure-instance) in k8s we just update
+  ;; the map instead of adding element to a set like in marathon's case
+  ([transient-store service-id instance-entry failure-id]
```

Comment:
I think we should not have this override and force the caller to provide a limit.
Also, the caller can assoc the `failure-id` into the instance.